### PR TITLE
Create sections 3, 4, 5, 6, and 7 of controller API guide

### PR DESCRIPTION
**Pull request by EMcWhinn**
Adding content for sections 3, 4, 5, 6, and 7

Create sections 3, 4, 5, 6, and 7 of the Automation Controller API Guide

https://issues.redhat.com/browse/AAP-21476

Affects `titles/controller-api-guide`

### CODE REVIEW

File: downstream/assemblies/platform/assembly-controller-api-filter.adoc [hunk 1]
```diff
@@ -2,4 +2,8 @@
 
 = Filtering in the API
 
-Any collection is what the system calls a “queryset” and can be filtered using various operators.
+Any collection is what the system calls a "queryset" that you can filter by using various operators.
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-controller-api-filter.adoc:5]: This is a bit of a strange sentence.

File: downstream/modules/platform/proc-controller-api-using-pagination.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+[literal, options="nowrap" subs="+attributes"]
+----
+{'count': 25, 'next': 'http://testserver/api/v2/some_resource?page=2', 'previous': None, 'results': [ ... ] }
+----
+
+.Procedure
+
+. Request the page given by the "next" sequential URL, to get to the next page.
+. Use the `page_size=XX` query string parameter to change the number of results returned for each request.
+** The `page_size` has a default maximum limit configured to 200, which is enforced when a user tries a value beyond it, for example, `?page_size=1000`.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-api-using-pagination.adoc:14]: configured to -> of

**Review by ianf77 — COMMENTED** [review]
A few comments

File: downstream/assemblies/platform/assembly-controller-api-filter.adoc (hunk 1 above)
Comment (EMcWhinn) [reply downstream/assemblies/platform/assembly-controller-api-filter.adoc:5]: I agree, what about:

"A collection is known by the system as a "queryset". You can filter this by using various operators."

Comment (ianf77) [reply downstream/assemblies/platform/assembly-controller-api-filter.adoc]: That'll do!

**Review by ianf77 — APPROVED** [review]

**Pull request merged**